api/src/permissions/modules/process-ast/utils/validate-path.ts

```diff
@@ -0,0 +1,48 @@
+import { ForbiddenError } from '@directus/errors';
+import type { Permission } from '@directus/types';
+
+// TODO add validation against schema object for non-existing fields and collections
+
+export function validatePath(path: string, permissions: Permission[], collection: string, fields: Set<string>) {
+	const pathSuffix = path === '' ? 'root' : `"${path}"`;
+
+	const permissionsForCollection = permissions.filter((permission) => permission.collection === collection);
+
+	if (permissionsForCollection.length === 0) {
+		throw new ForbiddenError({
+			reason: `You don't have permission to access collection "${collection}" or it does not exist. Queried in ${pathSuffix}.`,
+		});
```

Comment:
Even though this, and the other errors are really helpful, they do expose that the collection + field exists, as it differs from the standard `ForbiddenError` in case of a non-existent collection.

Reply:
This error returns the collections/fields that the user already provided in their query, so it shouldn't return any new information they didn't already provide. It merely shows you where in the query the wrong collection/field is, but doesn't expose whether that collection/field is real